src/compas_viewer/components/sceneform.py

```diff
@@ -101,42 +111,61 @@ def scene(self, scene: Scene):
+                    self.show_idx = i
+                    show_status = output[1]
+                    output = ""
+                strings.append(output)
 
             if node.parent.is_root:  # type: ignore
-                node.attributes["widget"] = QTreeWidgetItem(self, strings)  # type: ignore
+                widget = QTreeWidgetItem(self, strings)  # type: ignore
             else:
-                node.attributes["widget"] = QTreeWidgetItem(
+                widget = QTreeWidgetItem(
                     node.parent.attributes["widget"],
                     strings,  # type: ignore
                 )
+            widget.node = node
+            widget.setSelected(node.is_selected)
+            widget.setFlags(widget.flags() | Qt.ItemIsUserCheckable | Qt.ItemIsSelectable | Qt.ItemIsEnabled)  # Allow checkbox
 
-            node.attributes["widget"].node = node
-            node.attributes["widget"].setSelected(node.is_selected)
+            if self.show_idx is not None:
+                widget.setCheckState(self.show_idx, Qt.Checked if show_status else Qt.Unchecked)
 
             if self._backgrounds:
                 for col, background in self._backgrounds.items():
-                    node.attributes["widget"].setBackground(list(self.columns.keys()).index(col), QColor(*background(node).rgb255))
+                    widget.setBackground(list(self.columns.keys()).index(col), QColor(*background(node).rgb255))
+
+            node.attributes["widget"] = widget
 
+        self.adjust_column_widths()
         self._scene = scene
 
     def update(self):
-        from compas_viewer import Viewer
-
-        self.scene = Viewer().scene
+        self.scene = self.viewer.scene
 
-    def on_item_clickded(self):
-        selected_nodes = [item.node for item in self.selectedItems()]
-        for node in self.scene.objects:
-            node.is_selected = node in selected_nodes
-            if self.callback and node.is_selected:
-                self.callback(node)
+    def on_item_clicked(self, item, column):
+        if column == self.show_idx:
+            is_visible = item.checkState(self.show_idx) == Qt.Checked
+            item.node.show = is_visible
 
-        from compas_viewer import Viewer
+        if self.selectedItems():
+            self._selected_items = self.selectedItems()
+            selected_nodes = {item.node for item in self.selectedItems()}
+            for node in self.scene.objects:
+                node.is_selected = node in selected_nodes
+                if self.callback and node.is_selected:
+                    self.callback(node)
 
-        Viewer().renderer.update()
+        self.viewer.renderer.update()
 
     def on_item_selection_changed(self):
         for item in self.selectedItems():
             if self.callback:
                 self.callback(item.node)
+
+    def adjust_column_widths(self, item=None, column=None):
+        for i in range(self.columnCount()):
+            self.resizeColumnToContents(i)
```

Comment:
Would it be possible to shrink the column towards right side? like this:
<img width="272" alt="image" src="https://github.com/compas-dev/compas_viewer/assets/17893605/d2cce6c3-879f-401c-9a4d-c083b46c054f">